scripts/Tools/advanced-py-prof

```diff
@@ -0,0 +1,10 @@
+#! /bin/bash -e
+
+# Usage: advanced-py-prof <python exec and arguments>
+# Requires graphviz and gprof2dot
+
+DATE_STAMP=$(date "+%Y-%m-%d_%H%M%S")
+FILE_BASENAME=$(basename $1).${DATE_STAMP}
+python -m cProfile -o ${FILE_BASENAME}.prof "$@"
```

Comment:
Can you add code to make sure that these externals are in the path and exit gracefully if they do not?